public/app/core/components/AppChrome/ExtensionSidebar/ExtensionToolbarItem.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+
+import { Dropdown, Menu, ToolbarButton } from '@grafana/ui';
+
+import { useExtensionSidebarContext } from './ExtensionSidebarProvider';
+
+export function ExtensionToolbarItem() {
+  const [isOpen, setIsOpen] = useState(false);
+  const { components, dockedPluginId, setDockedPluginId } = useExtensionSidebarContext();
+
+  if (components.size === 0) {
+    return null;
+  }
+
+  if (components.size === 1) {
+    const component = components.values().next().value;
+    return (
+      <ToolbarButton
+        icon="web-section"
+        variant={dockedPluginId ? 'active' : 'default'}
+        tooltip={component?.meta.description}
+        onClick={() => {
+          setDockedPluginId(dockedPluginId === component?.meta.pluginId ? undefined : component?.meta.pluginId);
+        }}
+      />
+    );
+  }
+
+  const MenuItems = (
+    <Menu>
+      {Array.from(components.values()).map((c) => (
+        <Menu.Item
+          key={c.meta.pluginId}
+          active={dockedPluginId === c.meta.pluginId}
+          label={c.meta.title}
+          onClick={() => {
+            setDockedPluginId(dockedPluginId === c.meta.pluginId ? undefined : c.meta.pluginId);
+          }}
+        />
+      ))}
+    </Menu>
```

Comment:
Just for the future: I might make sense to group the extensions added by different plugins [similar to how we are doing it for Explore](https://github.com/grafana/grafana/blob/47f87171496f675bc44ddf839effaed1846bfa4e/public/app/features/explore/extensions/ToolbarExtensionPointMenu.tsx#L20). It could also make sense to add the logos of the plugins as icons to the `Menu.Item`s?